hyperspy/roi.py

```diff
@@ -1040,6 +1040,50 @@ def length(self):
         d_row, d_col = p1 - p0
         return np.hypot(d_row, d_col)
 
+    def angle(self, axis='horizontal', units='degrees'):
+        """"Angle between ROI line and selected axis
+
+        Parameters
+        ----------
+        axis : str, {'horizontal', 'vertical', 'x', 'y'}, optional
```

Comment:
'x' and 'y' can be names of axis and not necessarily the horizontal or vertical one of the signal axis. Maybe it could useful to be able to provide the name of the axis and remove these alias to avoid confusion?

Reply:
I'm not sure if I fully understood you here. Do you expect to provide name of the axis and 'horizontal'/'vertical'? I find 'horizontal/ 'vertical' convenient for using the method to align images with respect to image axis. (I also usually would not bother naming the image axes... )))

Reply:
Most reader will provide a name axis. I am suggesting to be able to use the name of the axis (the one from the axes_manager) as an alternative to 'horizontal' and 'vertical' and not to use 'x' and 'y' for the reason mentioned above.

Reply:
Removing 'x' and 'y' alias is no problem, and I'll do it to avoid confusion. Providing the axis name is actually a problem as `roi` objects have no 'axes_manager' and are not really connected to any signal as far as I understand? Am I right here or you have some suggestion how to implement it?

Reply:
The widgets should be `axes` aware. But if this is a bit too much of a pain then don't worry.

Reply:
I just have removed the alias. I would have to do quite a research to get something from widgets and that still this won't feel to me as the way to go, since the widgets are only used with `interactive()` (if I understand correctly)...